zboxcore/sdk/listworker_test.go

```diff
@@ -0,0 +1,294 @@
+		setup      func(t *testing.T, name string, p parameters, errMsg string)
+		wantErr    bool
+		errMsg     string
+	}{
+		{
+			name: "Test_Error_New_HTTP_Failed_By_Containing_" + string([]byte{0x7f, 0, 0}),
+			parameters: parameters{
+				blobberIdx:     41,
+				respStatusCode: 0,
+			},
+			setup:   func(t *testing.T, name string, p parameters, errMsg string) {},
+			wantErr: true,
+			errMsg:  `parse "Test_Error_New_HTTP_Failed_By_Containing_\u007f\x00\x00": net/url: invalid control character in URL`,
+		},
+		{
+			name: "Test_HTTP_Error",
+			parameters: parameters{
+				blobberIdx:     41,
+				respStatusCode: 0,
+			},
+			setup: func(t *testing.T, name string, p parameters, errMsg string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return strings.HasPrefix(req.URL.Path, name)
+				})).Return(&http.Response{
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte(""))),
+					StatusCode: p.respStatusCode,
+				}, fmt.Errorf(mockErrorMessage))
+			},
+			wantErr: true,
+			errMsg:  mockErrorMessage,
+		},
+		{
+			name: "Test_HTTP_Response_Failed",
+			parameters: parameters{
+				blobberIdx:     41,
+				respStatusCode: http.StatusInternalServerError,
+			},
+			setup: func(t *testing.T, name string, p parameters, errMsg string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return strings.HasPrefix(req.URL.Path, name)
+				})).Return(&http.Response{
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte(mockErrorMessage))),
+					StatusCode: p.respStatusCode,
+				}, nil)
+			},
+			wantErr: true,
+			errMsg:  "error from server list response: " + mockErrorMessage,
+		},
+		{
+			name: "Test_Error_HTTP_Response_Not_JSON_Format",
+			parameters: parameters{
+				blobberIdx:     41,
+				respStatusCode: http.StatusOK,
+			},
+			setup: func(t *testing.T, name string, p parameters, errMsg string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return strings.HasPrefix(req.URL.Path, name)
+				})).Return(&http.Response{
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte("this is not json format"))),
+					StatusCode: p.respStatusCode,
+				}, nil)
+			},
+			wantErr: true,
+			errMsg:  "list entities response parse error: invalid character 'h' in literal true (expecting 'r')",
+		},
+		{
+			name: "Test_Success",
+			parameters: parameters{
+				listHttpResp: listResponse{
+					ref: &fileref.Ref{
+						AllocationID: mockAllocationId,
+						Type:         mockType,
+					},
+				},
+				ListResult: fileref.ListResult{
+					AllocationRoot: mockAllocationRoot,
+					Meta: map[string]interface{}{
+						"type": mockType,
+					},
+				},
+				blobberIdx:     41,
+				respStatusCode: http.StatusOK,
+			},
+			setup: func(t *testing.T, testName string, p parameters, _ string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return req.URL.Path == "Test_Success"+zboxutil.LIST_ENDPOINT+mockAllocationTxId &&
```

Comment:
Both the "auth-token" and "path-hash" parameters are calculated by `getListInfoFromBlobber` so we should test that our function got it right.

Reply:
you can access the `params` using 
```go
mock.MatchedBy(func(req *http.Request) bool {
   query := req.URL.Query()
   require.EqualValue(t, query.Get("auth_token"), whatever autho_toekn should be)
   require.EqualValue(t, query.Get("path_hash), whatever pah_hash shoul be)
   return req.URL.Path == "Test_Success"+zboxutil.LIST_ENDPOINT+mockAllocationTxId && 
...
```